### PR TITLE
add methods to enable debug of SQL statements

### LINKED ISSUE #141: Simple way to debug queries
I would like to easily enable query logging like in gorm. 

e.g.

`store.Debug().Find(...)` could just log the executed query.

Or some method like `store.EnableQueryLog()` `store.DisableQueryLog()`

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #141 

- added Debug and DebugWith to Store
- added Debug and DebugWith to typed stores as well

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -51,6 +52,40 @@ func StoreFrom(to, from GenericStorer) {
 	to.SetGenericStore(from.GenericStore())
 }
 
+// LoggerFunc is a function that takes a log message with some arguments and
+// logs it.
+type LoggerFunc func(string, ...interface{})
+
+// debugProxy is a database proxy that logs all SQL statements executed.
+type debugProxy struct {
+	logger func(string, ...interface{})
```
Comment (dpordomingo) [inline store.go]: Why not using the already defined type `LoggerFunc` ?

File: store.go [hunk 2]
```diff
@@ -51,6 +52,40 @@ func StoreFrom(to, from GenericStorer) {
...
+	p.logger(fmt.Sprintf("kallax: Exec: %s", query), args...)
+	return p.proxy.Exec(query, args...)
+}
+
+func (p *debugProxy) Query(query string, args ...interface{}) (*sql.Rows, error) {
+	p.logger(fmt.Sprintf("kallax: Query: %s", query), args...)
+	return p.proxy.Query(query, args...)
+}
+
+func (p *debugProxy) QueryRow(query string, args ...interface{}) squirrel.RowScanner {
```
Comment (Serabe) [inline store.go]: Not tested

File: store.go [hunk 3]
```diff
@@ -51,6 +52,40 @@ func StoreFrom(to, from GenericStorer) {
...
+	p.logger(fmt.Sprintf("kallax: QueryRow: %s", query), args...)
+	return p.proxy.QueryRow(query, args...)
+}
+
+func (p *debugProxy) Prepare(query string) (*sql.Stmt, error) {
```
Comment (Serabe) [inline store.go]: Not tested.

Comment (erizocosmico) [inline store.go]: Can't test this one. There's no method that exposes a direct `Prepare` on Store. (It's done automatically underneath by the StmtCacher).

File: store.go (hunk 2 above)
Comment (erizocosmico) [inline store.go]: Same as `Prepare`

File: store.go (hunk 1 above)
Comment (erizocosmico) [inline store.go]: Totally true

File: store.go (hunk 2 above)
Comment (erizocosmico) [inline store.go]: Same as `Prepare`

**Comment by erizocosmico** [comment]
Done @dpordomingo

**Pull request merged**